### PR TITLE
Adds support for making a L2-projection of values

### PR DESCRIPTION
**Pull request by JimBrouzoulis**
Given Tensor{}-values determine the L2-projection of these onto the nodes of the mesh.

WIP
- Needs documentation and perhaps an example
- Can probably be made faster
- Increase test coverage

### CODE REVIEW

File: src/Grid/grid.jl [hunk 1]
```diff
@@ -81,13 +81,13 @@ end
 ##########################
 @inline getcells(grid::AbstractGrid) = grid.cells
 @inline getcells(grid::AbstractGrid, v::Union{Int, Vector{Int}}) = grid.cells[v]
-@inline getcells(grid::AbstractGrid, set::String) = grid.cells[grid.cellsets[set]]
+@inline getcells(grid::AbstractGrid, set::String) = grid.cells[collect(grid.cellsets[set])]
```
Comment (JimBrouzoulis) [inline src/Grid/grid.jl]: Bug: Not hit in tests

File: test/test_grid_dofhandler_vtk.jl [hunk 2]
```diff
@@ -94,3 +94,40 @@ end # of testset
 if OVERWRITE_CHECKSUMS
     close(csio)
 end
+
+
+# right = Vec{2, Float64}(ntuple(x->1.5, dim))
+# left = -right
+
+@testset "Grid utils" begin
```
Comment (JimBrouzoulis) [inline test/test_grid_dofhandler_vtk.jl]: Added some additional tests (to increase test coverage) when I added the `compute_vertex_values`

File: test/test_l2_projection.jl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    qp_values = [[f(spatial_coordinate(cv, qp, xe)) for qp in 1:getnquadpoints(cv)]]
+
+    # Now recover the nodal values using a L2 projection. Since f is quadratic and the interpolation as well, we should recover the exact nodal values
+    projector = L2Projector(cv, ip, grid)
+
+    point_vars = project(qp_values, projector)
+
+    ae = compute_vertex_values(grid, f)
+    # The projection gives the values in node order -> reorder ae
+    # @test point_vars[1:4] ≈ [ae[1], ae[2], ae[4], ae[3]]
```
Comment (JimBrouzoulis) [inline test/test_l2_projection.jl]: remove

**Comment by KristofferC** [comment]
Nice! No more cell to point data in paraview.

File: dev_cz.jl [hunk 4]
```diff
@@ -0,0 +1,285 @@
+using Pkg; Pkg.activate(".")
```
Comment (fredrikekre) [inline dev_cz.jl]: Whats this file yo

Comment (KristofferC) [reply dev_cz.jl]: Boomer

Comment (JimBrouzoulis) [reply dev_cz.jl]: Sneaky bonus code :) just for you guys ... no that one shouldn't be there.

File: docs/src/literate/l2_projection.jl [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# # L2-projection
+#
+# ![](heat_square_fluxes.png)
```
Comment (fredrikekre) [inline docs/src/literate/l2_projection.jl]: This image is missing :(

**Pull request merged**